src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +124,68 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String companyName} into an valid String.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given companyName is invalid.
+     */
+
+    public static String parseCompanyName(String companyName) throws ParseException {
+        requireNonNull(companyName);
+        String trimmedCompanyName = companyName.trim();
+        if (trimmedCompanyName.equals("")) {
+            throw new ParseException("Company name cannot be empty");
+        }
+        return trimmedCompanyName;
+    }
+
+    /**
+     * Parses a {@code String date} into an {@code Date}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+
+    public static Date parseDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmeddate = date.trim();
+        if (!Date.isValidDate(trimmeddate)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+        }
+        return new Date(trimmeddate);
+    }
+
+    /**
+     * Parses a {@code String time} into an {@code Time}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code time} is invalid.
+     */
+
+    public static Time parseTime(String time) throws ParseException {
+        requireNonNull(time);
+        String trimmedTime = time.trim();
+        if (!Time.isValidTime(trimmedTime)) {
+            throw new ParseException(Time.MESSAGE_CONSTRAINTS);
+        }
+        return new Time(time);
+    }
+
+    /**
+     * Parses a {@code String platform} into an {@code Platform}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code platform} is invalid.
```

Comment:
"platform" in these blocks of code should be changed to "location"

Reply:
fixed in 3549fd3